api/src/main/java/org/apache/iceberg/ManageSnapshots.java

```diff
@@ -80,4 +80,119 @@
    * wapId
    */
   ManageSnapshots cherrypick(long snapshotId);
+
+  /**
+   * Create a new branch pointing to the given snapshot id.
+   *
+   * @param name branch name
+   * @param snapshotId id of the snapshot which will be the head of the branch
+   * @return this for method chaining
+   * @throws IllegalArgumentException if a branch with the given name already exists
+   */
+  ManageSnapshots createBranch(String name, long snapshotId);
+
+  /**
+   * Create a new tag pointing to the given snapshot id
+   *
+   * @param name tag name
+   * @param snapshotId snapshotId for the head of the new branch.
+   * @return this for method chaining
+   * @throws IllegalArgumentException if a tag with the given name already exists
+   */
+  ManageSnapshots createTag(String name, long snapshotId);
+
+  /**
+   * Remove a branch by name
+   *
+   * @param name branch name
+   * @return this for method chaining
+   * @throws IllegalArgumentException if the branch does not exist
+   */
+  ManageSnapshots removeBranch(String name);
+
+  /**
+   * Rename a branch
+   *
+   * @param name name of branch to rename
+   * @param newName the desired new name of the branch
+   * @throws IllegalArgumentException if the branch to rename does not exist or if there is already a branch
+   * with the same name as the desired new name.
+   */
+  ManageSnapshots renameBranch(String name, String newName);
+
+  /**
+   * Remove the tag with the given name.
+   *
+   * @param name tag name
+   * @return this for method chaining
+   * @throws IllegalArgumentException if the branch does not exist
+   */
+  ManageSnapshots removeTag(String name);
+
+  /**
+   * Replaces the tag with the given name to point to the specified snapshot
+   *
+   * @param name Tag to replace
+   * @param snapshotId new snapshot id for the given tag
+   * @return this for method chaining
+   */
+  ManageSnapshots replaceTag(String name, long snapshotId);
+
+  /**
+   * Replaces the branch with the given name to point to the specified snapshot
+   *
+   * @param name Branch to replace
+   * @param snapshotId new snapshot id for the given branch
+   * @return this for method chaining
+   */
+  ManageSnapshots replaceBranch(String name, long snapshotId);
+
+  /**
+   * Replaces the branch with the given name to point to the source snapshot
+   *
+   * @param name Branch to replace
+   * @param source Source reference for the target to be replaced with
+   * @return this for method chaining
+   */
+  ManageSnapshots replaceBranch(String name, String source);
+
+  /**
+   * Performs a fast-forward of the given target branch up to the source snapshot if target is an ancestor of source
+   *
+   * @param name Branch to fast-forward
+   * @param source Source reference for the target to be fast forwarded to
+   * @return this for method chaining
+   * @throws IllegalArgumentException if the target branch is not an ancestor of source
+   */
+  ManageSnapshots fastForward(String name, String source);
```

Comment:
I think this should be `fastForwardBranch` because it makes no sense to "fast-forward" a tag, right?

Reply:
Good point, will update so it's fastForwardBranch. The source can be a tag (it can be any reference), but the actual reference we are fast-forwarding must be a branch.